lighthouse-core/formatters/critical-request-chains.js

```diff
@@ -99,39 +98,27 @@ class CriticalRequestChains extends Formatter {
     walk(tree, 0);
   }
 
-  static _getLongestChainLength(tree) {
-    let longestChain = 0;
-    this._traverse(tree, opts => {
-      const depth = opts.depth;
-      if (depth > longestChain) {
-        longestChain = depth;
-      }
-    });
-
-    // Always return the longest chain + 1 because the depth is zero indexed.
-    return (longestChain + 1);
-  }
-
-  static _getLongestChainDuration(tree) {
-    let longestChainDuration = 0;
+  /**
+   * Get stats about the longest initiator chain (as determined by time duration)
+   * @return {{duration: number, length: number, transferSize: number}}
+   */
+  static _getLongestChain(tree) {
+    const longest = {
+      duration: 0,
+      length: 0,
+      transferSize: 0
+    };
     this._traverse(tree, opts => {
       const duration = opts.chainDuration;
-      if (duration > longestChainDuration) {
-        longestChainDuration = duration;
+      if (duration > longest.duration) {
+        longest.duration = duration;
+        longest.transferSize = opts.chainTransferSize;
```

Comment:
this was outstanding, but why does this chainTransferSize not include the size of the file that pushed the chain out farthest? the duration is pulled from the youtube base.js in the screenshot which is ~430kb and larger than the entire chain size of ~390kb